src/nvim/os_unix.c

```diff
@@ -138,9 +138,9 @@ void mch_exit(int r) FUNC_ATTR_NORETURN
 {
   exiting = true;
 
-  if (p_title
-      && p_titlestring != NULL
-      && STRLEN(p_titlestring) == 0
+  if ((p_title
+       || (did_enable_title
+           && (p_titlestring == NULL || STRLEN(p_titlestring) == 0)))
       && p_titleold != NULL) {
```

Comment:
@nateozem why is `did_enable_title` needed? If 'title' is enabled and 'titleold' is non-empty, that should be enough. And we already check that anyways ...

Reply:
In the docs, it mention that if `titlestring` is not empty then `titleold` would be used.
```
       'titleold'		string	(default "Thanks for flying Vim")
 			global
 			{only available when compiled with the |+title|
 			feature}
 	This option will be used for the window title when exiting Vim if the
 	original title cannot be restored.  Only happens if 'title' is on or
 	'titlestring' is not empty.
```

In Vim 8.0, if `title` was ever on (even if turned back to off), and the `titlestring` is not empty, then the terminal title is changed to `titleold`. We can make things simple as you have pointed out, but we would need to change the description for `titleold`. The wording in the docs might be a mistake. I'm just not sure the thought process of the author.

Reply:
Please let me know if I'm reading it wrong. I do agree with you to check for only `title` and `titleold`.

Reply:
@nateozem I've made that change, try https://github.com/neovim/neovim/pull/7358

Reply:
Nice, I'll give it a try.